STYLE_GUIDE.md

```diff
@@ -0,0 +1,213 @@
+  names are only fine if they are used as temporary counters.
+  * Use camel-case with first character in lowercase for variables and
+  functions.
+  * Use camel-case with first character in uppercase for classes, structs and
+  namespaces.
+  * Use camel-case starting with an underscore, for private class members
+  (propertiesand methods).
+
+```C++
+// Incorrect
+class somedialog {
+    int h, w;
+
+public:
+    void event_handler();
+}
+
+// Correct
+class SomeDialog
+{
+    int _height;
+    int _width;
+
+public:
+    void eventHandler();
+}
+```
+  * Use all uppercase letters for naming macros.
+  * If a function has many arguments or if arguments do not fit on a single
+  line, each parameter should be given in different line and left-aligned.
+
+```C++
+// Incorrect
+int aFunctionWithLongArguments(firstArgumentIsLong, secondArgumentIsAlsoLong, thisOneExceededColumnLimit);
+
+// Correct
+int aFunctionWithLongArguments(firstArgumentIsLong,
+                               secondArgumentIsAlsoLong,
+                               thisOneExceededColumnLimit)
+```
+
+
+### Whitespaces
+  * Use blank lines to separate suitable groups of statements.
+  * Always use spaces between variables, operators and values.
+  * Always use a single space after a keyword and before a curly brace.
+
+```C++
+// Incorrect
+for(i=0;i<limit;++i){
+    int x=5+i*2;
+    magic(x);
+}
+
+// Correct
+for (i = 0; i < limit; ++i) {
+    int x = 5 + (i * 2);
+    magic(x);
+}
+```
+
+
+### Braces
+Use [K&R](https://en.wikipedia.org/wiki/Indentation_style#K&R_style) style for
+braces. Functions and classes have braces on a seaparate line of their own. For
+everything else use attached braces The opening brace goes on the same line as
+the start of the statement
+
+```C++
+// Incorrect
+void SomeDialog::eventHandler() {
+    doSomething();
+    doMore();
+
+    ...
+
+    if (somethingHappened)
+    {
+        doThis();
+    }
+    else
+    {
+        doThat();
+    }
+}
+
+// Correct
+void SomeDialog::eventHandler()
+{
+    doSomething();
+    doMore();
+
+    ...
+
+    if (somethingHappened) {
+        doThis();
+    } else {
+        doThat();
+    }
+}
+```
+
+Curly braces may only be used when the body of a conditional statement contains
+more than one line. Braces, however, _must_ be used if conditions themselves
+occupy multiple lines or there is an accompanying `else` statement to an `if`.
+
+```C++
+ // Fine
+if (someCondition)
+    return false;
+
+// Incorrect
+if (firstConditionThatTakesALotOfSpace
+    || secondConditionTakesALotOfSpaceAsWell)
+    return false;
+
+// Correct
+if (firstConditionThatTakesALotOfSpace
+    || secondConditionTakesALotOfSpaceAsWell) {
+    return false;
+}
+
+// Incorrect
+if (someCondition)
+    doSomething();
+else {
+    doSomeOtherThing();
+    return -1;
+}
+
+// Correct
+if (someCondition) {
+    doSomething();
+} else {
+    doSomeOtherThing();
+    return -1;
+}
+```
+
```

Comment:
Add a comment around whether having `if` condition on the same line as the statement within is acceptable or not.